<launch>

  <arg name="rviz" default="true" />

  <group ns="/ig/loam" >
    <rosparam command="load" file="$(find loam_velodyne)/config/ig_loam.yaml" />

    <node pkg="loam_velodyne" type="multiScanRegistration" name="multiScanRegistration" output="screen" >
      <param name="lidar" value="VLP-16" />
    </node>

    <node pkg="loam_velodyne" type="laserOdometry" name="laserOdometry" output="screen" respawn="false" />

    <node pkg="loam_velodyne" type="laserMapping" name="laserMapping" output="screen" />

    <node pkg="loam_velodyne" type="transformMaintenance" name="transformMaintenance" output="screen" />
  </group>


  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find loam_velodyne)/rviz_cfg/loam_velodyne.rviz" />
  </group>

</launch>
